web-gateway/src/main/java/controllers/TransactionController.java

```diff
@@ -77,28 +77,44 @@ public static Call transactionsPage(TransactionInfoStatus status, int page, int
         return requireUser(ctx(), user ->
                 loadNav(user).thenComposeAsync(nav -> {
-                    UUID itemId = UUID.fromString(id);
-                    CompletionStage<TransactionInfo> transactionFuture = transactionService.getTransaction(itemId).handleRequestHeader(authenticate(user)).invoke();
-                    return transactionFuture.handle((transaction, exception) -> {
-                        if (exception == null) {
-                            Optional<User> seller = Optional.empty();
-                            Optional<User> winner = Optional.empty();
-                            for (User u : nav.getUsers()) {
-                                if (transaction.getCreator().equals(u.getId())) {
-                                    seller = Optional.of(u);
-                                }
-                                if (transaction.getWinner().equals(u.getId())) {
-                                    winner = Optional.of(u);
-                                }
+                            UUID itemId = UUID.fromString(id);
+                            CompletionStage<TransactionInfo> transactionFuture = transactionService.getTransaction(itemId).handleRequestHeader(authenticate(user)).invoke();
+                    User user1;
+                            if (nav.getUser().isPresent()) {
+                         user1 = nav.getUser().get();
+                            }
+                            else{
+                                return  CompletableFuture.completedFuture(redirect(routes.UserController.createUserForm()));
                             }
-                            Currency currency = Currency.valueOf(transaction.getItemData().getCurrencyId());
-                            return ok(views.html.transaction.render(showInlineInstruction, Optional.of(transaction), user, seller, winner, Optional.of(currency), Optional.empty(), nav));
-                        } else {
-                            String msg = exception.getCause().getMessage();
-                            return ok(views.html.transaction.render(showInlineInstruction, Optional.empty(), user, Optional.empty(), Optional.empty(), Optional.empty(), Optional.of(msg), nav));
+                           return transactionFuture.handle((transaction, exception) -> {
+                                if (exception == null) {
+
+
+                                        Optional<User> seller = Optional.empty();
+                                        Optional<User> winner = Optional.empty();
+                                        if (transaction.getCreator().equals(user)) {
+
+                                            seller = Optional.of(user1);
+
+
+                                        }
+                                        if (transaction.getWinner().equals(user)) {
+
+                                            winner = Optional.of(user1);
+
+                                        }
```

Comment:
Similar to my comment above: this should be retrieving the seller and winner users, not just comparing them to the current user.